helix-core/src/main/java/org/apache/helix/task/RuntimeJobDag.java

```diff
@@ -198,11 +198,15 @@ public void generateJobList() {
     resetJobListAndDependencyMaps();
     computeIndependentNodes();
     _readyJobList.addAll(_independentNodes);
-    if (_isJobQueue && _readyJobList.size() > 0) {
+    if (_isJobQueue && !_readyJobList.isEmpty()) {
```

Comment:
Not quite sure whether this is just a style change? If we dont change it, would it cause any problem?

Reply:
ArrayDeque.size has a nontrivial implementation (https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/ArrayDeque.java#L649) but this is an emptiness check.

Reply:
Little did I know! I think this is good.